scout/adapter/mongo/query.py

```diff
@@ -126,29 +125,14 @@ def build_variant_query(
         if query.get("cohorts"):
             mongo_case_query["cohorts"] = {"$in": query["cohorts"]}
 
-        if mongo_case_query != {}:
-            mongo_case_query["collaborators"] = institute_id
+        if mongo_case_query != {} or institute_ids:
```

Comment:
This is the real fix: instead of querying for all variants in all cases and then check if user is allowed to see the cases in gene_variants controller, query directly for variants in the cases that user has access to